### PR TITLE
Emon tx hub with dc hall class

### PR DESCRIPTION
**Pull request by Britnell**
Just a few changes during verification of this version running. 
Precise VREF measured and referenced as 3.311 V

### CODE REVIEW

File: Atmega328/emonPi_RFM69CW_RF12Demo_DiscreteSampling/HallSensor.cpp [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+ *    
+ */
+
+void HallSensor::Initialise( float in_VREF ){
+  
+  analogReference(DEFAULT);   // EXTERNAL
+   
+  VREF = in_VREF;   //3.283;
+    
+  Readings = 3;
```
Comment (jlark) [inline Atmega328/emonPi_RFM69CW_RF12Demo_DiscreteSampling/HallSensor.cpp:37]: just a best practice (not urgent) is to have constants in all caps lock.

File: Atmega328/emonPi_RFM69CW_RF12Demo_DiscreteSampling/emonPi_RFM69CW_RF12Demo_DiscreteSampling.ino [hunk 2]
```diff
@@ -268,10 +276,27 @@ void setup()
...
+   *   DC current hall sensor ---- SETUP
+   *    ---------------------------------------------------
+   *    
+   *    - give precise voltage of 3.3V Ref on emonhubTx
+   *    - Hall Offset variable is for calibrating other sensors towards 
+   *      the * reference sensor. Offset seems to be mostly linear compared to other sensors
+   *    - Debugging boolean turns on serial debugging messages of adc value, voltage 
+   *      and current translation over serial. This confuses BBB
+   */
+  HALL_VREF = 3.311;                // Precise voltage reference for ADC
```
Comment (jlark) [inline Atmega328/emonPi_RFM69CW_RF12Demo_DiscreteSampling/emonPi_RFM69CW_RF12Demo_DiscreteSampling.ino:290]: `HAL_VREF` - can't this be measured?

File: Atmega328/emonPi_RFM69CW_RF12Demo_DiscreteSampling/emonPi_RFM69CW_RF12Demo_DiscreteSampling.ino [hunk 3]
```diff
@@ -451,8 +478,27 @@ void loop()
 
    emonPi.v_battery_bank=vin;
 
+  // ######################   HALL  SENSOR  #########
```
Comment (jlark) [inline Atmega328/emonPi_RFM69CW_RF12Demo_DiscreteSampling/emonPi_RFM69CW_RF12Demo_DiscreteSampling.ino:481]: Thanks for the extra comments :+1:

**Comment by jlark** [comment]
ok some quick simple feedback else looks good. Do we need the HEX files as well?

**Comment by Britnell** [comment]
I have commited and pushed more changes with corections.

**Comment by jlark** [comment]
LGTM

**Pull request merged**